app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java

```diff
@@ -0,0 +1,43 @@
+package ingsw.group1.findmyphone.contacts;
+
+import android.widget.SearchView;
+
+/**
+ * Class implements {@link SearchView.OnQueryTextListener}.
+ * This is a listener to activate the search on the contact list through a {@link SMSContactRecyclerAdapter}.
+ *
+ * @author Giorgia Bortoletti
+ */
+public class SMSContactSearchListener implements SearchView.OnQueryTextListener {
+
+    private SMSContactRecyclerAdapter recyclerAdapter;
+
+    public SMSContactSearchListener(SMSContactRecyclerAdapter recyclerAdapter) {
+        this.recyclerAdapter = recyclerAdapter;
+    }
+
+    /**
+     * It is invoked when user submits the text to search.
+     *
+     * @param query     Text to search
+     *
+     * @return false, as default
```

Comment:
Please consider adding a brief explanation on what the value `false` does, because it is not clear.

Reply:
This method is invoked by default when the user adds o removes a letter from the search bar. I don't call it somewhere. Therefore I don't save or view the value that returns. I wrote about 'default' because nothing determines true or false. 
I don't know how explain it. Do you have any suggestion?

Reply:
When you are not sure about descriptions in overridden methods try to look at the parent method, as it will probably contain well detailed descriptions. For example, if you look at `onQueryTextSubmit(...)` you will find this info:
```java
/**
* Called when the user submits the query. This could be due to a key press on the
* keyboard or due to pressing a submit button.
* The listener can override the standard behavior by returning true
* to indicate that it has handled the submit request. Otherwise return false to
* let the SearchView handle the submission by launching any associated intent.
*
* @param query the query text that is to be submitted
*
* @return true if the query has been handled by the listener, false to let the
* SearchView perform the default action.
*/
boolean onQueryTextSubmit(String query);
```
So you can modify your specification accordingly:
```suggestion
     * @return false, letting the SearchView perform the default action.
```
The same thing can be done also in the method below.